### PR TITLE
Adds Navigation Menu based on Radix

### LINKED ISSUE #240: Add component for NavMenu
Based on Radix [NavMenu](https://www.radix-ui.com/docs/primitives/components/navigation-menu).

The triggers should have variants for (at least) primary, secondary and brand (could also do tertiary).
You should be able to reuse the styling from the buttons.

Add animation (like in the example first then we can tweak to our design).
You could have different Callout variants if you want.

Add a wrapped version to the AppBar that then uses the brand styling in the triggers.

- [ ]  Change implementation
- [ ]  Test 
- [ ]  Add to storybook - include version with react router links.

### PR DESCRIPTION
**Pull request by kristian-committed**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Bug fix
[ x ] Enhancement
[ ] Documentation
[ ] Deployment
[ ] Process (e.g. improve continuous integration)
[ ] Other, please explain:

**Which issue (if any) does this pull request address?**

Fixes #240

Screenshot 2022-03-15 at 17 01 54
Screenshot 2022-03-15 at 17 32 40
Screenshot 2022-03-15 at 14 29 56
Screenshot 2022-03-15 at 17 01 42

### CODE REVIEW

File: src/components/NavigationMenu/NavigationMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,430 @@
...
+
+const StyledList = styled(List, {
+  all: 'unset',
+  display: 'flex',
+  justifyContent: 'center',
+  alignItems: 'stretch',
+  padding: 4,
+  listStyle: 'none',
+  fontSize: '$0',
+  borderRadius: '$3',
```
Comment (stuarthendren) [inline src/components/NavigationMenu/NavigationMenu.tsx:73]: Unless we have a really good reason all the borderRadius should be set to `'$default'`.

The valid token values come from the `stitches.config.ts` file property:

```
    radii: {
      default: '2px',
      round: '50%',
      pill: '9999px',
    },
```

So $3 here is not valid - and probably doesn't do anything.
Where you have used 4 below, that probably works but again I think we should at least try it with $default first - remember we are not trying to emulate the radix style but to fit with our style.

This also applies to the other properties too. So, for example we should not use 4 in the above padding, as this is not one of our design tokens.

File: src/components/NavigationMenu/NavigationMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,430 @@
...
+  from: { opacity: 1 },
+  to: { opacity: 0 },
+})
+
+const StyledMenu = styled(Root, {
+  position: 'relative',
+  display: 'flex',
+  justifyContent: 'center',
+  width: '100%',
+  zIndex: 1,
```
Comment (stuarthendren) [inline src/components/NavigationMenu/NavigationMenu.tsx:62]: Is this required?
If so, we might want to add a token in the stitches config under zIndex for it - e.g. $nav, or $menu. However, I don't expect it to be required.

File: src/components/NavigationMenu/NavigationMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,430 @@
...
+  borderRadius: '$3',
+  backgroundColor: '$transparent',
+  border: 'none',
+})
+
+const itemStyles = {
+  padding: '$2 $3',
+  outline: 'none',
+  userSelect: 'none',
+  borderRadius: 4,
```
Comment (stuarthendren) [inline src/components/NavigationMenu/NavigationMenu.tsx:82]: use token

File: src/components/NavigationMenu/NavigationMenu.tsx [hunk 4]
```diff
@@ -0,0 +1,430 @@
...
+  '&:focus': { position: 'relative', boxShadow: `0 0 0 $0 $colors$primary` },
```
Comment (stuarthendren) [inline src/components/NavigationMenu/NavigationMenu.tsx:83]: use a shadow token - or if this is much better add it to the shadow tokens - don't forget to check how it looks in light and dark.

File: src/components/NavigationMenu/NavigationMenu.tsx [hunk 5]
```diff
@@ -0,0 +1,430 @@
...
+const ContentListItem = forwardRef<
+  ElementRef<typeof StyledLink>,
+  ComponentProps<typeof StyledLink>
+>(({ children, ...props }, forwardedRef) => (
+  <ListItem>
+    <StyledLink
+      {...props}
+      ref={forwardedRef}
+      css={{
+        borderRadius: 3,
```
Comment (stuarthendren) [inline src/components/NavigationMenu/NavigationMenu.tsx:306]: use token

**Review by stuarthendren — CHANGES_REQUESTED** [review]

**Comment by stuarthendren** [comment]
Can I have new screen shots, in light and dark - don't have time to run it up here.

**Comment by kristian-committed** [comment]
Screenshot 2022-03-17 at 10 37 11
Screenshot 2022-03-17 at 10 37 28
Screenshot 2022-03-17 at 10 37 55
Screenshot 2022-03-17 at 10 38 10

File: src/components/NavigationMenu/NavigationMenu.tsx [hunk 6]
```diff
@@ -0,0 +1,430 @@
...
+
+  display: 'flex',
+  flexDirection: 'column',
+  gap: '$2',
+
+  '&::before': {
+    boxShadow: 'none',
+  },
+  filter:
+    'drop-shadow(0px 3px 2.5px rgba(0,0,0,0.2)) drop-shadow(0px 6px 5px rgba(0,0,0,0.14)) drop-shadow(0px 1px 9px rgba(0,0,0,0.12))',
```
[the remark this answers is not in the document]
Comment (kristian-committed) [reply src/components/NavigationMenu/NavigationMenu.tsx]: Didn't seem to be able to but as this filter doesn't add much, I've since removed it

**Comment by stuarthendren** [comment]
can you add a small gap between the buttons ($2 probably).
And I'm a bit confused why the secondary button in the lower rows is larger the than the others. It shouldn't be.
I'm confused as it doesn't happen in the top rows. So that might imply it's something to do with the icon though I first thought it would be the border.

**Comment by kristian-committed** [comment]
> can you add a small gap between the buttons ($2 probably). And I'm a bit confused why the secondary button in the lower rows is larger the than the others. It shouldn't be. I'm confused as it doesn't happen in the top rows. So that might imply it's something to do with the icon though I first thought it would be the border.

Added the gap and fixed sizing issue, turns out it was still the border causing the issue, the no-icon buttons are sized 100% to match up besides the icon ones so it wasn't noticeable. border-box doesn't seem to work how I expected, so I've added a transparent border to the non-secondary styles so that size is consistent.

**Review by stuarthendren — APPROVED** [review]

**Pull request merged**